merino/curated_recommendations/sections.py

```diff
@@ -136,6 +146,10 @@ async def get_corpus_sections(
         sections[cs.externalId] = map_corpus_section_to_section(
             cs, rank, is_legacy_section=cs.externalId in legacy_sections
         )
+    for sname, section in sections.items():
+        for r in section.recommendations:
+            if r.corpusItemId in sid_map:
+                r.update_scheduled_corpus_item_id(sid_map.get(r.corpusItemId) or "")
```

Comment:
```suggestion
                r.update_scheduled_corpus_item_id(sid_map[r.corpusItemId])
```
The check `r.corpusItemId in sid_map` guarantees the key is present. We don't want to send empty strings as `scheduledCorpusItemId`.